linkerd/app/inbound/src/lib.rs

```diff
@@ -181,14 +197,24 @@ impl<A: OrigDstAddr> Config<A> {
                     http_profile_route_proxy.into_inner(),
                 ))
                 .into_new_service()
-                .push_on_response(
-                    svc::layers()
-                        .push_lock()
-                        .push(metrics.stack.layer(stack_labels("profile"))),
-                )
                 // Caches profile stacks.
-                .check_new_clone_service::<Profile>()
-                .spawn_cache(cache_capacity, cache_max_idle_age)
+                .check_new_service_routes::<Profile, Target>()
+                .cache(
+                    svc::layers().push_on_response(
+                        svc::layers()
+                            // TODO: self-evicting buffers
+                            // // If the service has been ready & unused for `cache_max_idle_age`,
+                            // // fail it.
+                            // .push_idle_timeout(cache_max_idle_age)
+                            // // If the service has been unavailable for an extend time, eagerly
+                            // // fail requests.
+                            // .push_failfast(Duration::from_secs(10))
+                            // // Shares the service, ensuring discovery errors are propagated.
+                            // .push_spawn_buffer(10, Duration::from_secs(1))
+                            .push_lock()
```

Comment:
before this merges, i'd like to figure out why tests fail when this changes

Reply:
@olix0r what was the answer to this?